Database throws panic if initialized multiple times

We get a panic when using the DB in tests.

```
panic: duplicate metrics collector registration attempted [recovered]
        panic: duplicate metrics collector registration attempted

goroutine 82 [running]:
testing.tRunner.func1.2({0x12dac40, 0xc0004001a0})
        /home/agaffney/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.24.1.linux-amd64/src/testing/testing.go:1734 +0x21c
testing.tRunner.func1()
        /home/agaffney/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.24.1.linux-amd64/src/testing/testing.go:1737 +0x35e
panic({0x12dac40?, 0xc0004001a0?})
        /home/agaffney/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.24.1.linux-amd64/src/runtime/panic.go:792 +0x132
github.com/prometheus/client_golang/prometheus.(*Registry).MustRegister(0x2073e20, {0xc005dc08a0?, 0x13f8bea?, 0x21?})
        /home/agaffney/go/pkg/mod/github.com/prometheus/client_golang@v1.22.0/prometheus/registry.go:406 +0x65
github.com/prometheus/client_golang/prometheus.MustRegister(...)
        /home/agaffney/go/pkg/mod/github.com/prometheus/client_golang@v1.22.0/prometheus/registry.go:177
github.com/blinklabs-io/dingo/database/plugin/blob/badger.(*BlobStoreBadger).registerBlobMetrics(0x1?)
        /home/agaffney/dev/blinklabs-io/dingo/database/plugin/blob/badger/metrics.go:97 +0xa96
```

--- a/database/database.go
+++ b/database/database.go
@@ -21,6 +21,7 @@ import (
 
 	"github.com/blinklabs-io/dingo/database/plugin/blob"
 	"github.com/blinklabs-io/dingo/database/plugin/metadata"
+	"github.com/prometheus/client_golang/prometheus"
 )
 
 // Database represents our data storage services
@@ -94,13 +95,14 @@ func (d *Database) init() error {
 // New creates a new database instance with optional persistence using the provided data directory
 func New(
 	logger *slog.Logger,
+	promRegistry prometheus.Registerer,
 	dataDir string,
 ) (*Database, error) {
-	metadataDb, err := metadata.New("sqlite", dataDir, logger)
+	metadataDb, err := metadata.New("sqlite", dataDir, logger, promRegistry)
 	if err != nil {
 		return nil, err
 	}
-	blobDb, err := blob.New("badger", dataDir, logger)
+	blobDb, err := blob.New("badger", dataDir, logger, promRegistry)
 	if err != nil {
 		return nil, err
 	}

--- a/database/database_test.go
+++ b/database/database_test.go
@@ -42,7 +42,7 @@ func TestInMemorySqliteMultipleTransaction(t *testing.T) {
 		}
 		return nil
 	}
-	db, err := database.New(nil, "") // in-memory
+	db, err := database.New(nil, nil, "") // in-memory
 	if err != nil {
 		t.Fatalf("unexpected error: %s", err)
 	}

--- a/database/plugin/blob/badger/database.go
+++ b/database/plugin/blob/badger/database.go
@@ -26,6 +26,7 @@ import (
 
 	"github.com/blinklabs-io/dingo/database/plugin"
 	badger "github.com/dgraph-io/badger/v4"
+	"github.com/prometheus/client_golang/prometheus"
 )
 
 // Register plugin
@@ -40,22 +41,25 @@ func init() {
 
 // BlobStoreBadger stores all data in badger. Data may not be persisted
 type BlobStoreBadger struct {
-	dataDir   string
-	db        *badger.DB
-	logger    *slog.Logger
-	gcEnabled bool
+	dataDir      string
+	db           *badger.DB
+	logger       *slog.Logger
+	promRegistry prometheus.Registerer
+	gcEnabled    bool
 }
 
 // New creates a new database
 func New(
 	dataDir string,
 	logger *slog.Logger,
+	promRegistry prometheus.Registerer,
 ) (*BlobStoreBadger, error) {
 	var blobDb *badger.DB
 	var err error
 	db := &BlobStoreBadger{
-		dataDir: dataDir,
-		logger:  logger,
+		dataDir:      dataDir,
+		logger:       logger,
+		promRegistry: promRegistry,
 	}
 	if dataDir == "" {
 		// No dataDir, use in-memory config
@@ -108,7 +112,9 @@ func (d *BlobStoreBadger) init() error {
 		d.logger = slog.New(slog.NewJSONHandler(io.Discard, nil))
 	}
 	// Configure metrics
-	d.registerBlobMetrics()
+	if d.promRegistry != nil {
+		d.registerBlobMetrics()
+	}
 	// Configure GC
 	if d.gcEnabled {
 		go d.blobGc(time.NewTicker(5 * time.Minute))

--- a/database/plugin/blob/badger/metrics.go
+++ b/database/plugin/blob/badger/metrics.go
@@ -94,5 +94,5 @@ func (d *BlobStoreBadger) registerBlobMetrics() {
 			),
 		},
 	)
-	prometheus.MustRegister(collector)
+	d.promRegistry.MustRegister(collector)
 }

--- a/database/plugin/blob/store.go
+++ b/database/plugin/blob/store.go
@@ -19,6 +19,7 @@ import (
 
 	badgerPlugin "github.com/blinklabs-io/dingo/database/plugin/blob/badger"
 	badger "github.com/dgraph-io/badger/v4"
+	"github.com/prometheus/client_golang/prometheus"
 )
 
 type BlobStore interface {
@@ -35,6 +36,7 @@ type BlobStore interface {
 func New(
 	pluginName, dataDir string,
 	logger *slog.Logger,
+	promRegistry prometheus.Registerer,
 ) (BlobStore, error) {
-	return badgerPlugin.New(dataDir, logger)
+	return badgerPlugin.New(dataDir, logger, promRegistry)
 }

--- a/database/plugin/metadata/sqlite/database.go
+++ b/database/plugin/metadata/sqlite/database.go
@@ -27,6 +27,7 @@ import (
 	"github.com/blinklabs-io/dingo/database/plugin"
 	"github.com/blinklabs-io/dingo/database/plugin/metadata/sqlite/models"
 	"github.com/glebarez/sqlite"
+	"github.com/prometheus/client_golang/prometheus"
 	"gorm.io/gorm"
 	gormlogger "gorm.io/gorm/logger"
 	"gorm.io/plugin/opentelemetry/tracing"
@@ -44,16 +45,18 @@ func init() {
 
 // MetadataStoreSqlite stores all data in sqlite. Data may not be persisted
 type MetadataStoreSqlite struct {
-	dataDir     string
-	db          *gorm.DB
-	logger      *slog.Logger
-	timerVacuum *time.Timer
+	dataDir      string
+	db           *gorm.DB
+	logger       *slog.Logger
+	promRegistry prometheus.Registerer
+	timerVacuum  *time.Timer
 }
 
 // New creates a new database
 func New(
 	dataDir string,
 	logger *slog.Logger,
+	promRegistry prometheus.Registerer,
 ) (*MetadataStoreSqlite, error) {
 	var metadataDb *gorm.DB
 	var err error
@@ -101,9 +104,10 @@ func New(
 		}
 	}
 	db := &MetadataStoreSqlite{
-		db:      metadataDb,
-		dataDir: dataDir,
-		logger:  logger,
+		db:           metadataDb,
+		dataDir:      dataDir,
+		logger:       logger,
+		promRegistry: promRegistry,
 	}
 	if err := db.init(); err != nil {
 		// MetadataStoreSqlite is available for recovery, so return it with error

--- a/database/plugin/metadata/store.go
+++ b/database/plugin/metadata/store.go
@@ -23,6 +23,7 @@ import (
 	"github.com/blinklabs-io/gouroboros/ledger"
 	lcommon "github.com/blinklabs-io/gouroboros/ledger/common"
 	ochainsync "github.com/blinklabs-io/gouroboros/protocol/chainsync"
+	"github.com/prometheus/client_golang/prometheus"
 	"gorm.io/gorm"
 )
 
@@ -236,6 +237,7 @@ type MetadataStore interface {
 func New(
 	pluginName, dataDir string,
 	logger *slog.Logger,
+	promRegistry prometheus.Registerer,
 ) (MetadataStore, error) {
-	return sqlite.New(dataDir, logger)
+	return sqlite.New(dataDir, logger, promRegistry)
 }

--- a/internal/node/load.go
+++ b/internal/node/load.go
@@ -47,7 +47,7 @@ func Load(cfg *config.Config, logger *slog.Logger, immutableDir string) error {
 		)
 	}
 	// Load database
-	db, err := database.New(logger, cfg.DatabasePath)
+	db, err := database.New(logger, nil, cfg.DatabasePath)
 	if err != nil {
 		return err
 	}

--- a/node.go
+++ b/node.go
@@ -77,7 +77,7 @@ func (n *Node) Run() error {
 	}
 	// Load database
 	dbNeedsRecovery := false
-	db, err := database.New(n.config.logger, n.config.dataDir)
+	db, err := database.New(n.config.logger, n.config.promRegistry, n.config.dataDir)
 	if db == nil {
 		n.config.logger.Error(
 			"failed to create database",

